src/main/fc/fc_msp.c

```diff
@@ -824,6 +824,25 @@ static bool mspFcProcessOutCommand(uint16_t cmdMSP, sbuf_t *dst, mspPostProcessF
         sbufWriteU8(dst, currentBatteryProfile->capacity.unit);
         break;
 
+    case MSP2_INAV_MISC2:
+        // Timers
+        sbufWriteU32(dst, micros() / 1000000); // On time (seconds)
+        sbufWriteU32(dst, getFlightTime()); // Flight time (seconds)
+
+        // Throttle
+        int16_t thr = 0;
+        const int minThrottle = getThrottleIdleValue();
+        bool autoThrottle = navigationIsControllingThrottle();
+        if(autoThrottle)
+            thr = (constrain(rcCommand[THROTTLE], PWM_RANGE_MIN, PWM_RANGE_MAX) - PWM_RANGE_MIN) * 100 / (PWM_RANGE_MAX - PWM_RANGE_MIN);
+        else
+            thr = (constrain(rcCommand[THROTTLE], PWM_RANGE_MIN, PWM_RANGE_MAX ) - minThrottle) * 100 / (motorConfig()->maxthrottle - minThrottle);
```

Comment:
Can't we read it from the Mixer somehow? I do not like the idea of computing this again here in the MSP part

Reply:
Ok, I moved the computing from OSD to Mixer and created a new variable to store the value to be used anywhere. Then I use the new variable on OSD and on the new MSP message.